modules/core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/SessionImpl.java

```diff
@@ -1491,26 +1493,32 @@ public boolean isFulltextStoredInBlob() {
     }
 
     @Override
-    public Map<String, String> getBinaryFulltext(Serializable id) {
+    public Map<String, String> getBinaryFulltext(Serializable id, Document doc) {
         if (fulltextDescriptor.getFulltextDisabled()) {
             return null;
         }
         RowId rowId = new RowId(Model.FULLTEXT_TABLE_NAME, id);
         Map<String, String> map = mapper.getBinaryFulltext(rowId);
         String fulltext = map.get(BINARY_FULLTEXT_MAIN_KEY);
         if (fulltextDescriptor.getFulltextStoredInBlob() && fulltext != null) {
-            // fulltext is actually the blob  key
-            // now retrieve the actual fulltext from the blob content
-            DocumentBlobManager blobManager = Framework.getService(DocumentBlobManager.class);
-            try {
-                BlobInfo blobInfo = new BlobInfo();
-                blobInfo.key = fulltext;
-                Blob blob = blobManager.readBlob(blobInfo, getRepositoryName());
-                fulltext = blob.getString();
-                map.put(BINARY_FULLTEXT_MAIN_KEY, fulltext);
-            } catch (IOException e) {
-                throw new PropertyException("Cannot read fulltext blob for doc: " + id, e);
+            if (doc == null) {
+                // could not find doc (shouldn't happen)
+                fulltext = null;
```

Comment:
It can happen if fulltext is not stored in blob, no?

Reply:
No, there's a `if (fulltextDescriptor.getFulltextStoredInBlob()` just before.